tests/psa-client-server/psasim/Makefile

```diff
@@ -1,23 +1,51 @@
-CFLAGS ?= -Wall -Werror -std=c99 -D_XOPEN_SOURCE=1 -D_POSIX_C_SOURCE=200809L
+CFLAGS += -Wall -Werror -std=c99 -D_XOPEN_SOURCE=1 -D_POSIX_C_SOURCE=200809L
 
 ifeq ($(DEBUG),1)
-	CFLAGS += -DDEBUG -O0 -g
+	CFLAGS += -DDEBUG
```

Comment:
Why remove `-O0` and `-g` - if we are going to debug, they are very useful options to have built the executables with!

Reply:
I know I know, I've used them thousand times in the past, but those are passed in by `all.sh`

Reply:
Yes, but this is specifically for a `DEBUG` build (the test on line 3). Why require a developer to modify a file when these can just be kept in?

Reply:
I was afraid that adding extra `-O0 -g` on a line which already contained `-O2` (from `ASAN_FLAGS`) could not work properly, but googling a bit turned out that it actually does what we expect. I fixed this in https://github.com/Mbed-TLS/mbedtls/pull/9121/commits/c98f8ab5f7879fd1a42ba325f7909c109443b691